src/components/CreateAccount.js

```diff
@@ -96,24 +96,36 @@ class CreateAccount extends Component {
     }
 
     componentDidUpdate() {
-        const { step, query } = this.state;
+        const { step } = this.state;
         if (step === 'created') {
             if (this.isWhistle()) {
                 window.postMessage('whistle_signup_complete');
             } else {
-                const urlParameters =
-                    query &&
-                    Object.keys(query)
-                        .map(param => `${param}=${query[param]}`)
-                        .join('&');
+                const redirectUri = this.getRedirect();
                 setTimeout(() => {
-                    window.location.href = `${
-                        window.config.DEFAULT_REDIRECT_URI
-                    }?${urlParameters}`;
+                    window.location.href = redirectUri;
                 }, 5000);
             }
         }
     }
+    getRedirect = () => {
+        const { username } = this.state;
+        const { location: { query } } = this.props;
```

Comment:
I think you want props.query here, not location.query which is part of the react client side routing.

Reply:
`location.query` will yield us the query params on the current url.
`props.query` doesn't appear to have any values assigned to it, although it is referenced when checking if the user is using whistle.

I'm not entirely familiar with the codebase, but it appears we would want `location.query`.

Reply:
No, we want the query that is persisted in db not the current query. Point being that an user enter with `?foo=bar` on the first step and exit with `?foo=bar` when the account is created (which could be in a completely different browser)